components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/auth/OAuthHandler.java

```diff
@@ -0,0 +1,214 @@
+        }
+        String remoteURLString = properties.get(EntitlementEndpointConstants.PROPERTY_NAME_AUTH_SERVER);
+        if (remoteURLString != null) {
+            remoteServiceURL = remoteURLString;
+        } else {
+            remoteServiceURL = LOCAL_AUTH_SERVER;
+        }
+        userName = properties.get(EntitlementEndpointConstants.PROPERTY_NAME_USERNAME);
+        password = properties.get(EntitlementEndpointConstants.PROPERTY_NAME_PASSWORD);
+    }
+
+    private String getOAuthAuthzServerURL() {
+        if (remoteServiceURL != null && !remoteServiceURL.endsWith("/")) {
+            remoteServiceURL += "/";
+        }
+        return remoteServiceURL;
+    }
+
+    private OAuth2ClientApplicationDTO validateAccessToken(String accessTokenIdentifier)
+            throws Exception {
+
+        // if it is specified to use local authz server (i.e: local://services)
+        if (remoteServiceURL.startsWith(LOCAL_PREFIX)) {
+            OAuth2TokenValidationRequestDTO oauthValidationRequest = new OAuth2TokenValidationRequestDTO();
+            OAuth2TokenValidationRequestDTO.OAuth2AccessToken accessToken = oauthValidationRequest.new OAuth2AccessToken();
+            accessToken.setTokenType(OAuthServiceClient.BEARER_TOKEN_TYPE);
+            accessToken.setIdentifier(accessTokenIdentifier);
+            oauthValidationRequest.setAccessToken(accessToken);
+
+            OAuth2TokenValidationService oauthValidationService = new OAuth2TokenValidationService();
+            OAuth2ClientApplicationDTO oauthValidationResponse = oauthValidationService
+                    .findOAuthConsumerIfTokenIsValid(oauthValidationRequest);
+
+            return oauthValidationResponse;
+        }
+        // else do a web service call to the remote authz server
+        try {
+            ConfigurationContext configContext = ConfigurationContextFactory
+                    .createConfigurationContextFromFileSystem(null, null);
+            OAuthServiceClient oauthClient = new OAuthServiceClient(getOAuthAuthzServerURL(),
+                    userName, password, configContext);
+            org.wso2.carbon.identity.oauth2.stub.dto.OAuth2ClientApplicationDTO validationResponse;
+            validationResponse = oauthClient.findOAuthConsumerIfTokenIsValid(accessTokenIdentifier);
+
+            OAuth2ClientApplicationDTO appDTO = new OAuth2ClientApplicationDTO();
+            appDTO.setConsumerKey(validationResponse.getConsumerKey());
+
+            OAuth2TokenValidationResponseDTO validationDto = new OAuth2TokenValidationResponseDTO();
+            validationDto.setAuthorizedUser(validationResponse.getAccessTokenValidationResponse()
+                    .getAuthorizedUser());
+            validationDto
+                    .setValid(validationResponse.getAccessTokenValidationResponse().getValid());
+            appDTO.setAccessTokenValidationResponse(validationDto);
+            return appDTO;
+        } catch (AxisFault axisFault) {
```

Comment:
This catch and throw does not seems to add a value. Should include a meaningful message when throwing.